### PR TITLE
[processing] Add a dedicated pan tool for navigating models

### LINKED ISSUE #40714: [Modeler] Regression : no hand tool in 3.16.1
**Describe the bug**
In 3.10.3 we can navigate in modeler. 
In 3.16.1 select tool is enable by default and we could not navigate in model making navigation in complex model difficult

**How to Reproduce**

1. Go to 'Processing => Model > Right click > edit model'
2. Try to navigate in modeler

3.16.1 : Select tool by default and no hand tool in bar navigation
image

**QGIS and OS versions**

Version de QGIS | 3.16.1-Hannover | Branche de code de QGIS | Release 3.16
-- | -- | -- | --
Compilé avec Qt | 5.15.2 | Utilisant Qt | 5.15.2
Compilé avec GDAL/OGR | 3.0.4 | Utilisé avec GDAL/OGR | 3.0.4
Compilé avec GEOS | 3.8.0-CAPI-1.13.1 | Utilisé avec GEOS | 3.8.0-CAPI-1.13.1
Compilé avec SQLite | 3.33.0 | Fonctionne avec SQLite | 3.33.0
Version du client PostgreSQL | 12.5 | Version de SpatiaLite | 4.3.0a
Version de QWT | 6.1.5 | Version de QScintilla2 | 2.11.5
Compilé avec PROJ | 6.3.2 | Fonctionne avec PROJ | Rel. 6.3.2, May 1st, 2020
Version de l'OS | Manjaro Linux
Extensions Python actives | quick_map_services; slyr_community; SemiAutomaticClassificationPlugin; lizmap; Generalizer3; pdaltools; plugin_reloader; flight_planner; MagicWand-master; qgis_resource_sharing; ale; VoGisProfilTool; qconsolidate3; GroupStats; profiletool; SpreadsheetLayers; pluginbuilder3; OSMDownloader; DataPlotly; QGISSortAndNumber-master; LDMP; processing; MetaSearch; db_manager

**Additional context**

Please add select and hand tool

Checklist before submitting

- [x] Search through existing issue reports and gis.stackexchange.com to check whether the issue already exists
- [x] Test with a [clean new user profile](https://docs.qgis.org/testing/en/docs/user_manual/introduction/qgis_configuration.html?highlight=profile#working-with-user-profiles).
- [x] Create a light and self-contained sample dataset and project file which demonstrates the issue : Doesn't depend on dataset

### PR DESCRIPTION
**Pull request by nyalldawson**
Fixes #40714

### CODE REVIEW

**Comment by roya0045** [comment]
Thanks a ton! I was just getting used to constantly zoom in and out on my bigger models.

File: src/gui/processing/models/qgsmodelviewtoolpan.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/***************************************************************************
+                             qgsmodelviewtoolpan.cpp
+                             ------------------------------------
+    Date                 : March 2020
```
Comment (nirvn) [inline src/gui/processing/models/qgsmodelviewtoolpan.cpp:4]: Did you see on this one for that long? :wink:

**Review by nirvn — APPROVED** [review]

File: src/gui/processing/models/qgsmodelviewtoolpan.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/gui/processing/models/qgsmodelviewtoolpan.cpp:4]: Bah. Oh well, not worth the travis rebuild time to fix :stuck_out_tongue:

**Comment by nyalldawson** [comment]
Manually copied to https://github.com/qgis/QGIS/pull/40712

**Pull request merged**